modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/io/Clients.scala

```diff
@@ -69,6 +70,19 @@ object Clients {
       .withRequestTimeout(readTimeout)
       .withMaxTotalConnections(maxConnections)
       .resource
+      .map(Retry[F](retryPolicy, redactHeadersWhen))
+
+  private def retryPolicy[F[_]] =
+    RetryPolicy[F](
+      RetryPolicy.exponentialBackoff(maxWait = 10.seconds, maxRetry = 10),
```

Comment:
Do I understand right that we think the first retry will have a very high likelihood of success, even if we retry after no delay?  And am I right in the old implementation of blaze client, it would retry immediately without a delay?

I checked [the http4s implementation of exponentialBackoff](https://github.com/http4s/http4s/blob/v0.21.33/client/src/main/scala/org/http4s/client/middleware/Retry.scala#L175-L179).  It looks like the first retry happens after ~1000 milliseconds (with random factor), and it's not configurable.  That's _probably_ OK if it's a rare exception.  But I wonder if we can do better to avoid adding latency to the output of enrich.  Is it worth considering if we should use a custom retry policy that retries almost immediately?

Reply:
I agree, we should try to retry immediately.